### ISSUE TITLE
Replace commands by certain text

### ISSUE THREAD

**Issue by tobiasdiez** [valentjn/vscode-ltex#43]
**Is your feature request related to a problem? Please describe.**
Some commands produce text, for example `\Latex` or my custom command `\ie` which expands to `i.e.`. In these cases, it would be nice to specify the replacement text to allow ltex to pick up the correct sentence structure.

**Describe the solution you'd like**
Extend the dummy replacement map to allow to specify the replacement. For example:
```
ltex.commands.dummy = {
    "\\cite{}", // no explicit dummy text
    "\ie" : "i.e." // explicit dummy text
}
```

**Describe alternatives you've considered**
Use normal dummy replacement. This works for example for `\LaTeX` but not for commands that don't produce substantives.

**Comment by valentjn** [comment]
@tobiasdiez Thanks for the feature request. Could you please give a real-life example/sentence in which replacing your command with a dummy results in a false positive/negative diagnostic?

It's true that this won't work for verbs, but I'm curious where the need to have this actually comes up. Both "i.e." and "\cite" work with dummies (at least for my citation style).

(Edit: One example I already came across is a noun that starts with a vowel, but I'm not sure if that use case is so common that it would justify the introduction of a new setting.)

**Comment by tobiasdiez** [comment]
One example where this comes up is singular vs plural.
For example, languagetools complains about `The dummies barks loudly.` but not about `The dummy barks loudly.`
Thus replacing `textcites` by "dummy" leads to no error in ̀ The \textcites{AbbatiCirelliEtAl1986,AbbatiCirelliEtAl1989} barks loudly.` although it should, as textcites is always producing a list of references.

Concerning the `i.e.`: Using the definition `\ie = i.e.`, there is a comma missing after "\ie" in the sentence `This is a test, \ie an experiment`. See https://community.languagetool.org/rule/show/IE_NO_COMMA?lang=en&subId=1

(In general, I agree, it appears that a custom dummy replacement is only necessary in rare cases.)

**Comment by valentjn** [comment]
I don't get a diagnostic for `This is a test, i.e. an experiment.` either. So the mechanism seems to be working; replacing it would make no difference. Directly using `\ie` would currently result in `\ie` being omitted from the checked text (since it's an unknown command) and also produce no diagnostics. Am I missing something?

If it's only about plural citations, that's something that could be fixed specifically for those commands like `\textcites` by changing the dummy. I don't think an extra setting would be necessary if there's no more use cases.

**Comment by tobiasdiez** [comment]
I think the above i.e. rule is only available in 5.0-snapshot, thus not yet released. 

Otherwise the singular vs plural issue is probably the most relevant one. Adding a new configuration might be the most flexible solution (to support custom commands like `\dog` vs `\dogs`) but is perhaps also a bit of an overkill. I let you decide ;-)

**Comment by tobiasdiez** [comment]
Thanks for the quick fix once again!

**Comment by valentjn** [comment]
@tobiasdiez You're welcome. Fix released in 5.0.0.
